src/main/java/hudson/plugins/active_directory/CacheConfiguration.java

```diff
@@ -0,0 +1,41 @@
+package hudson.plugins.active_directory;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+/**
+ * Cache configuration
+ */
+public class CacheConfiguration<K,V,E extends Exception> {
+    private final int size;
+    private final int ttl;
+
+    /**
+     * CacheConfiguration DataBoundConstructor
+     *
+     * @param size of the cache in terms of number of elements
+     * @param ttl of the cache in terms of seconds
+     */
+    @DataBoundConstructor
+    public CacheConfiguration(int size, int ttl) {
+        this.size = Math.max(0, Math.min(size, 1000));
+        this.ttl = Math.max(0, Math.min(ttl, 3600));
+    }
+
+    /**
+     * Get size of the cache
+     *
+     * @return the size of the cache
+     */
+    public int getSize() {
+        return size;
+    }
+
+    /**
+     * Get TTL of the cache
+     *
+     * @return the ttl of the cache in seconds
+     */
+    public int getTtl() {
+        return ttl;
+    }
+}
```

Comment:
I would move the cache construction here (using a static factory method, so that you can account on the null and easily type-inference the generics)